frontend/src/components/Wallet.js

```diff
@@ -0,0 +1,184 @@
+import {
+	Button,
+	Container,
+	Table,
+	TableBody,
+	TableCell,
+	TableContainer,
+	TableRow,
+} from "@mui/material";
+import React, { useEffect } from "react";
+import Grid from "@mui/material/Grid";
+import ConditionalAlert from "./ConditionalAlert";
+
+function AccountEntry({ data }) {
+	let balance = "NaN";
+	// data.balance can have NaN value if no Metamask or other Wallet installed
+	if (!isNaN(data.balance)) {
+		balance = utils.formatEther(data.balance);
+	}
+	
+	return <TableRow>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>
+			{data.pubAddress}
+		</TableCell>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>
+			{balance}
+		</TableCell>
+	</TableRow>;
+}
+
+function EthereumAccountDetails ({ publicAddresses, setErrorNoMetamaskMessage }) {
+	const [detailData, setDetailData] = useState([]);
+
+	async function fetchAccountsDetails(publicAddresses) {
+		// TODO: publicAddresses is still not loaded. 
+		// Solve temporarily by waiting for 0.5 sec.
+		await delayMsec(500);
+		const provider = await detectEthereumProvider();
+		let details = [];
+		if (!provider) {
+			setErrorNoMetamaskMessage("Couldn't get data from Ethereum Provider. Please install MetaMask!");
+			console.error("no ethereum provider");
+			// Display only ETH address from backend
+			publicAddresses.map(pubAddress => {
+				details.push({ "pubAddress": pubAddress, "balance": NaN });
+			}) ;
+			setDetailData(details);
+			return;
+		}
+		const length = publicAddresses.length;
+		for (let i=0; i<length; i++) {
+			let pubAddress = publicAddresses[i];
+			const balance = await provider.request({ 
+				method: "eth_getBalance", 
+				params: [pubAddress, "latest"] 
+			});	// "Wei" balance in hex format, like 0x5a4e1804f198f1d99.
+			if (balance) {
+				details.push({ 
+					pubAddress: pubAddress,
+					balance: balance
+				});
+			}
+		}
+		setDetailData(details);
+	}
+
+	useEffect(() => {
+		fetchAccountsDetails(publicAddresses);
+	}, []);
+
+	return <TableContainer style={{ color: "white" }}>
+		<Table>
+			<TableBody>
+				<TableRow>
+					<TableCell style={{ color: "white", borderBottom: "none" }}>
+						<Typography variant="subtitle1" gutterBottom component="div">
+							Account (Public Address)
+						</Typography>
+					</TableCell>
+					<TableCell style={{ color: "white", borderBottom: "none" }}>
+						<Typography variant="subtitle1" gutterBottom component="div">
+							Balance (Eth)
+						</Typography>
+					</TableCell>
+				</TableRow>
+				{detailData.map((value, key) => {
+					return <AccountEntry key={key} data={value} />;
+				})}
+			</TableBody>
+		</Table>
+	</TableContainer>;
+}
+
+/**
+ * Component to display and add Eth address from Wallet. Currently only Metamask is supported.
+ * @param user user object of the currently logged in user/admin
+ * @returns {JSX.Element}
+ */
+export default function Wallet({ user }) {
+	const [loadedAddresses, setLoadedAddresses] = useState(false);
```

Comment:
Variable's name sounds like an array to me instead of a boolean. I would prefer, if variable names of a boolean variable would sound like a predicate or question. A common convention of naming booleans is using `is` or `has` as name prefix or `Flag` as suffix.